monix-eval/shared/src/test/scala/monix/eval/TaskGatherNSuite.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2014-2019 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.eval
+
+import monix.execution.exceptions.DummyException
+import monix.execution.internal.Platform
+
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+object TaskGatherNSuite extends BaseTestSuite {
+
+  test("Task.gatherN should execute in parallel for async tasks") { implicit s =>
+    val seq = Seq(
+      Task.evalAsync(1).delayExecution(1.second),
+      Task.evalAsync(2).delayExecution(2.seconds),
+      Task.evalAsync(3).delayExecution(3.seconds)
+    )
+    val f = Task.gatherN(1)(seq).runToFuture
```

Comment:
- This isn't actually checking parallelism with N = 1
- It should execute in parallel even if Tasks aren't async to begin with.
- `delayExecution` already makes things async, btw.